src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1451,7 +1452,12 @@ public synchronized void restartLeaderElection(QuorumVerifier qvOLD, QuorumVerif
     }
 
     public String getNextDynamicConfigFilename() {
-        return configFilename + QuorumPeerConfig.nextDynamicConfigFileSuffix;
+        if (configFilename != null) {
```

Comment:
Update pull request based on offline chat with Alex. The idea is to detect the null cases for configFileName and for those cases don't write intermediate reconfig file to disk because if we do, the shared "null.dynamic.cfg" file will be overwritten and might likely break other things without notice.